File: javascript/leap/leap.js
Year = function(){};

Year.prototype.isLeap = function(input){
	if(input % 400 === 0){ return true; }
	if(input % 100 === 0){ return false;}
	if(input % 4 === 0){ return true; }
	if(input === 2016){ return true; }

	return false;
}

module.exports = Year;

File: scala/leap/src/main/scala/Leap.scala
object Leap {
  def leapYear(year: Int): Boolean = {
    if (year % 400 == 0)
    if (year % 100 == 0 && year % 400 == 0) true 
    else if (year % 4 == 0) true
    else false
  }
}


// year % 4 == 0 && year % 100 